docs/src/references/synchronize-daemons.md

```diff
@@ -0,0 +1,116 @@
+# Introduction
+This document explains the scenarios that require the startup of different tedge daemons to be synchronized,
+and solutions to avoid this synchronization by relying on dynamic runtime health status.
+
+# Problem statement
+
+Thin-edge.io has three main issues for synchronizing the daemons
+
+1. The thin-edge daemons depend on another daemon's liveness to delegate or request some work.
+ For example, tedge-mapper-c8y depends on tedge-agent. When the tedge-mapper-c8y is started,
+ the mapper sends a request to get the existing software list on `tedge/commands/req/software/list` topic,
+ if the tedge-agent is not up and running then the request message will be lost. So, it's better to check if the agent
```

Comment:
I agree, to make sure the `tedge-agent` was up at least once, we must change the agent to declare the supported operations `device-restart and software` when it starts for the first time, now the tedge-mapper can pick this and send it across to the cloud. This will make the tedge-agent to be in line with the other plugins as well.